### PR TITLE
Make quantifier module classes derive from EnvObj

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/theory/quantifiers/sygus/sygus_module.h [hunk 1]
```diff
@@ -49,7 +50,7 @@ class QuantifiersInferenceManager;
  * Modules implement an initialize function, which determines whether the module
  * will take responsibility for the given conjecture.
  */
-class SygusModule
+class SygusModule : protected EnvObj
```
Comment (aniemetz) [inline src/theory/quantifiers/sygus/sygus_module.h:53]: It looks like this doesn't actually use anything from `Env`. Do we have to derive here?

File: src/theory/quantifiers/quant_module.h [hunk 2]
```diff
@@ -40,7 +41,7 @@ class TermDb;
  * This is the virtual class for defining subsolvers of the quantifiers theory.
  * It has a similar interface to a Theory object.
  */
-class QuantifiersModule
+class QuantifiersModule : protected EnvObj
```
Comment (aniemetz) [inline src/theory/quantifiers/quant_module.h:44]: Ditto, this doesn't use anything from `Env`?

File: src/theory/quantifiers/sygus/sygus_module.h (hunk 1 above)
Comment (ajreynol) [reply src/theory/quantifiers/sygus/sygus_module.h:53]: This inheritance in part allows `Rewriter::rewrite` to be replaced by `rewrite` throughout the majority of the quantifiers code.

File: src/theory/quantifiers/quant_module.h (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/quant_module.h:44]: Ditto.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**